cytosnake/utils/cyto_paths.py

```diff
@@ -0,0 +1,184 @@
+"""
+module: cyto_paths.py
+
+This module will contain functions that handles `cytosnake's` pathing
+"""
+
+from typing import Optional
+from pathlib import Path
+
+# cytosnake imports
+import cytosnake
+from cytosnake.utils.file_utils import file_search, find_project_dir
+from cytosnake.guards.path_guards import is_valid_path
+
+
+def get_meta_path() -> Path:
+    """returns meta path configurational file path.
+
+    Returns
+    -------
+    Path
+        Path object pointing to `_paths.yaml` config file
+    """
+
+    # getting project directory
+    proj_dir = find_project_dir()
+    if not proj_dir.exists():
+        raise FileNotFoundError("Current directory is not a project folder")
+
+    return proj_dir.absolute()
+
+
+# TODO: this should be placed in the guards module
+def is_cytosnake_dir(dir_path: Optional[str | Path] = None) -> bool:
+    """Checks if the current directory has been set up for cytosnake. Searches
+    for the `.cytosnake` file in current or specified directory.
+
+    Parameters
+    ----------
+    dir_path : Optional[str | Path]
+        Path to directory. If None, current directory will be used.
+        [Default=None]
+
+    Returns
+    -------
+    bool
+        True if directory has been initialized for cytosnake, else False
+    """
+
+    # setting to current directory if not path is specified
+    if dir_path is None:
+        dir_path = Path().absolute()
+
+    # type checking
+    if not is_valid_path(dir_path):
+        _type = type(dir_path)
+        raise ValueError(f"dir_path must be either str or Path type, not: {_type}")
+    elif isinstance(dir_path, str):
+        dir_path = Path(str).absolute()
+
+    # get project older
+    cyto_proj_dir = dir_path / ".cytosnake"
+    return bool(cyto_proj_dir.exists())
+
+
+def get_cytosnake_package_path() -> Path:
+    """Returns paths where the package is installed
+
+    Return
+    ------
+    Path
+        Returns absolute path of where the package is installed
+
+    Raises
+    ------
+    FileNotFoundError
+        Raised if the `.git` folder is not found.
+    """
+
+    # get location of this file
+    # -- check if he ".git" folder exists
+    project_path = Path(cytosnake.__path__[0]).parent
+    git_path = project_path / ".git"
```

Comment:
this function seems to require cytosnake to run in a git configured directory. Is this necessary? I think we will always use cytosnake in a github repo, but I don't think it will be universally true. Am I not thinking of the reason why `.git` is necessary?

Reply:
I am using `.git` as a landmark to find where the project folder is installed. I guess it's a hacky way to identify where the package is installed in your environment. if there is a more practical way in how to do this I am all ears :)

Reply:
Yeah, it does seem hacky. Maybe ok for now?

As I see it, cytosnake will not work if used outside of a github repo. I think we want to support non-git usage.

Could `init` have some sort of `--project_folder` command argument that defaults to `pwd` (or `.`)?

Reply:
I think there is some confusion. The `get_cytosnake_package_path()` returns the absolute path of `CytoSnake` within the python env. For example, after pip installing, we want the path where `CytoSnake` is installed in the python env.

Reply:
but returning the absolute path of Cytosnake requires there to be a `.git` folder?

Reply:
Yeah. I have not figure out a way on how to get the package path without using `.git` folder. I was doing some reading in using the `__name__` can help resolve this.